src/plugins/timeseries-plugin.ts

```diff
@@ -0,0 +1,259 @@
+import { OffscreenRenderEngine } from '../engines/offscreen-render-engine';
+import { SeparatedInteractionsEngine } from '../engines/separated-interactions-engine';
+import { HitRegion, RegionTypes } from '../types';
+import UIPlugin from './ui-plugin';
+
+export type TimeseriesPoint = [number, number];
+
+interface TimeseriesPointsSummary {
+    min: number;
+    max: number;
+    first: number;
+    last: number;
+}
+
+export type TimeseriesPluginStyles = {
+    defaultHeight?: number;
+    padding?: number;
+    color?: string;
+};
+
+type TimeseriesPluginStylesNoOptional = {
+    defaultHeight: number;
+    padding: number;
+    color: string;
+};
+
+export const defaultCPUPluginStyles: TimeseriesPluginStylesNoOptional = {
+    defaultHeight: 68,
+    padding: 5,
+    color: 'red',
+};
+
+type TimeseriesPointWithIndex = [idx: number, ts: number, value: number, normalizedTs: number, normalizedValue: number];
+
+enum TimeseriesPointWithIndexX {
+    idx = 0,
```

Comment:
til now our enum members were always uppercased
not sure to what we should stick with, but it should be consistent

Reply:
will happily change

Reply:
done